internal/validation/validation.go

```diff
@@ -0,0 +1,39 @@
+package validation
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+func NormalizeAndValidatePath(path string) (string, error) {
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return "", fmt.Errorf("invalid directory: %s", err)
+	}
+
+	exists, err := dirExists(absPath)
+	if err != nil {
+		return "", fmt.Errorf("invalid directory: %s", err)
+	}
+
+	if !exists {
+		return "", fmt.Errorf("directory does not exist: %s", absPath)
+	}
+
+	return absPath, nil
+}
+
+func dirExists(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+
```

Comment:
nit: this function looks a little too spaced out visually. I'd probably strip the empty lines except after if blocks to match the convention you used in `NormalizeAndValidatePath`